### PR TITLE
Document fields for Schema Transforms

### PR DESCRIPTION
**Pull request by pabloem**
r: @manavgarg

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by manavgarg — APPROVED** [review]
Looks good. Thanks.

**Comment by pabloem** [comment]
Run Java PreCommit

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java [hunk 1]
```diff
@@ -157,24 +160,41 @@ public static Builder builder() {
           .Builder();
     }
 
+    @SchemaFieldDescription(
+        "The bigquery table to write to. Format: [${PROJECT}:]${DATASET}.${TABLE}")
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java:164]: BigQuery

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java [hunk 2]
```diff
@@ -157,24 +160,41 @@ public static Builder builder() {
...
     public abstract String getTable();
 
+    @SchemaFieldDescription(
+        "Optional field that specifies whether the job is allowed to create new tables. "
+            + "The following values are supported: CREATE_IF_NEEDED (the job may create the table), CREATE_NEVER ("
+            + "the job must fail if the table does not exist already).")
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java:170]: Can we add a comment about the default? I think this was it (based on https://cloud.google.com/bigquery/docs/reference/rest/v2/Job), but we should probably test this too. 

The default value is CREATE_IF_NEEDED.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java [hunk 3]
```diff
@@ -157,24 +160,41 @@ public static Builder builder() {
...
     @Nullable
     public abstract String getCreateDisposition();
 
+    @SchemaFieldDescription(
+        "Specifies the action that occurs if the destination table already exists. "
+            + "The following values are supported: "
+            + "WRITE_TRUNCATE (overwrites the table data), "
+            + "WRITE_APPEND (append the data to the table), "
+            + "WRITE_EMPTY (job must fail if the table is not empty).")
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java:179]: Could we also add the default here too if the value isn't specified? I believe it should be WRITE_APPEND based on this PR: https://github.com/apache/beam/pull/25434

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteReadSchemaTransformProvider.java [hunk 4]
```diff
@@ -137,14 +150,29 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
+    @SchemaFieldDescription(
+        "The schema in which the data is encoded in the Kafka topic. "
+            + "For AVRO data, this is a schema defined with AVRO schema syntax "
+            + "(https://avro.apache.org/docs/1.10.2/spec.html#schemas). "
+            + "For JSON data, this is a schema defined with JSON-schema syntax (https://json-schema.org/).")
     public abstract String getSchema();
 
+    @SchemaFieldDescription(
+        "The GCP project where the Pubsub Lite reservation resides. This can be a "
+            + "project number of a project ID.")
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteReadSchemaTransformProvider.java:167]: or a project ID.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteReadSchemaTransformProvider.java [hunk 5]
```diff
@@ -137,14 +150,29 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
     public abstract @Nullable String getProject();
 
+    @SchemaFieldDescription(
+        "The name of the subscription to consume data. This will be concatenated with "
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteReadSchemaTransformProvider.java:171]: The name of the subscription from which to consume data.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java [hunk 6]
```diff
@@ -141,12 +144,22 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
   @AutoValue
   @DefaultSchema(AutoValueSchema.class)
   public abstract static class PubsubLiteWriteSchemaTransformConfiguration {
+    @SchemaFieldDescription(
+        "The GCP project where the Pubsub Lite reservation resides. This can be a "
+            + "project number of a project ID.")
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java:149]: or a project ID.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java [hunk 7]
```diff
@@ -141,12 +144,22 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
     public abstract String getProject();
 
+    @SchemaFieldDescription("The region or zone where the Pubsub Lite reservation resides.")
     public abstract String getLocation();
 
+    @SchemaFieldDescription(
+        "The name of the topic to publish data into. This will be concatenated with "
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java:156]: to publish data to.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java [hunk 8]
```diff
@@ -141,12 +144,22 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
+            + "the project and location parameters to build a full topic path.")
     public abstract String getTopicName();
 
+    @SchemaFieldDescription(
+        "The encoding format for the data stored in Pubsub Lite. Valid options are: "
+            + SUPPORTED_FORMATS_STR)
```
Comment (mahagen) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java:162]: Same comment here, I think the comma in the format string will need a space after it.

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformConfiguration.java [hunk 9]
```diff
@@ -56,24 +60,47 @@ public static Builder builder() {
...
+          + " here for bootstrapping—this list only impacts the initial hosts used to discover the full set"
+          + " of servers. This list should be in the form `host1:port1,host2:port2,...`")
   public abstract String getBootstrapServers();
 
   @Nullable
   public abstract String getConfluentSchemaRegistryUrl();
 
-  // TODO(pabloem): Make data format an ENUM
+  @SchemaFieldDescription(
+      "The encoding format for the data stored in Kafka. Valid options are: " + VALID_FORMATS_STR)
```
Comment (mahagen) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformConfiguration.java:74]: commas in formats str should have spaces after them.

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProvider.java [hunk 10]
```diff
@@ -139,12 +142,25 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
+        "The encoding format for the data stored in Kafka. Valid options are: "
+            + SUPPORTED_FORMATS_STR)
     public abstract String getFormat();
 
     public abstract String getTopic();
 
+    @SchemaFieldDescription(
+        "A list of host/port pairs to use for establishing the initial connection to the"
+            + " Kafka cluster. The client will make use of all servers irrespective of which servers are specified"
+            + " here for bootstrapping—this list only impacts the initial hosts used to discover the full set"
```
Comment (mahagen) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProvider.java:155]: space around the dash

**Review by mahagen — COMMENTED** [review]
lg!

**Pull request merged**